### PR TITLE
WIP: Modernize layer to use new primitives

### PR DESCRIPTION
**Pull request by marcoceppi**
This is much like battlemidget/juju-layer-nginx#6 but pushes even further on the idea. I made sure `rubylib` is available for backwards compat, but ultimately this will make things work as most others do today.

The last thing I want to do is build a custom tactic that helps warn/error for those using dependency.txt in their charm so that people can update without problem

### CODE REVIEW

**Comment by marcoceppi** [comment]
@battlemidget this takes some liberal opinions, while I wrap up the last bits I'm curious of feedback.

File: tactics/ruby.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
+from charmtools.build.tactics import ExactMatch, Tactic
+
+
+class DependenciesTxtTactic(ExactMatch, Tactic):
+    FILENAME = 'dependencies.txt'
+    kind = "dynamic"
+
+    def __call__(self):
+        print('DIE FUCKER')
```
Comment (mbruzek) [inline tactics/ruby.py:10]: ProTip: remove profanity from pull requests.

Comment (adam-stokes) [reply tactics/ruby.py:10]: feel free to open a new PR for the profanity, or leave it for flair.

**Comment by marcoceppi** [comment]
This was not ready to merge

**Comment by adam-stokes** [comment]
Yea sorry, please re-open, I didnt see the WIP

**Pull request merged**